beat/heart.py

```diff
@@ -2254,7 +2319,71 @@ def get_phase_arrival_time(engine, source, target, wavename=None, snap=True):
         deltat = 1. / store.config.sample_rate
         atime = trace.t2ind(atime, deltat, snap=round) * deltat
     return atime
+##Mahdi
+arrival_catalog = {'p': ('any_P', 'Z'),
+                   'sv':('any_SV', 'ZR'),
+                   'sh':('any_SH', 'T')}
+class PolarityDataset(object):
+    
+        
+    def __init__(self, store, polarityconfig, data_path):
+
+        self.dataset = num.array([])
+        self.targets = []    
+        self.config = polarityconfig
+        self.store = store
+
+        stations = utility.load_objects(data_path)[0]
+        station_names = num.array([str(station.split(" ")[0]) for station in polarityconfig.stations_polarities])            
+        crust_ind = 0
+        
+        phase = arrival_catalog[polarityconfig.name.split("wfarrival")[0].lower()][0]
+        if phase == 'any_P':
+            if 'moho' in [disc.name for disc in self.store.config.earthmodel_1d.discontinuities()]:
+                definition = 'p,P,p\\,P\\,Pv_(cmb)p'
+            else:
+                definition = 'p,P,p\\,P\\'
+            phases = gf.TPDef(id='any_P', definition=definition)
+
+        for i, station in enumerate(stations):
+            if station.station in station_names:
+                self.targets.append(PolarityTarget(
+                                codes=(station.network, station.station,'%i' % crust_ind, '%s' % polarityconfig.channels),  # n, s, l, c
+                                lat=station.lat, lon=station.lon, azimuth=station.azimuth*d2r, distance=station.dist_deg,
+                                elevation=float(station.elevation), store_id=self.store.config.id, 
+                                phase=phases))
+
+        if polarityconfig.binary_input:
+            self.dataset = num.array([station.azimuth for station in stations])
```

Comment:
seems inconsistent here, below dataset is then polarities, here it is azimuths, still you have the double definition of "dataset", the dataset object shouldnt have a dataset attribute, rather name it what it is- polarities ...

Reply:
We haven't developed a binary dataset, I just wrote something to be developed later! not a real code.

Reply:
ok then just remove it for now